### ISSUE TITLE
KeyTracking component not being cleaned up 

### ISSUE THREAD

**Issue by francesar** [codephil-columbia/typephil#118]
Seems that the KeyTracking component doesn't get cleaned up after you exit the page.
<img width="1062" alt="Screen Shot 2019-06-28 at 3 59 58 PM" src="https://user-images.githubusercontent.com/15842531/60375375-d43ac200-99bd-11e9-9062-26c3686d1abf.png">
As seen in the image on the left, the `endGames` function gets called even after we have logged out and redirected to the login page and all the properties for the KeyTracking component are null. Ideally, we'd have everything for this component cleaned up when we redirect to a different page. Additionally, there is `setTimeout` in this function that could be contributing to this issue.

Let's talk about how we can fix this :)

**Comment by francesar** [comment]
I'm not super sure what the point of the `setTimout` is here, but I think an easy fix for this that keeps the structure of what you already have is to set an instance var on `KeyTracking` to be the output of `setTimeout`. By doing this, you can manually call `clearTimeout` on the timer in `KeyTracking`'s `componentWillUnmount.`

**Comment by sangjunp4rk** [comment]
let's talk about this tonight

**Comment by mattvanegas13** [comment]
think i've done it for the game - would either of you like to double check, @sangjunp4rk @francesar ?

**Comment by francesar** [comment]
@mattvanegas13 I still had an issue with this, unrelated to the keytracking cleanup you just added. The  counterPage's interval was still not being cleaned up all the way when I started a game and immediately logged out. I ran it throw a debugger and I think it should be good now. Besides that, everything looks good ty for the fix! 

@mattvanegas13 @sangjunp4rk pls double check that what I added works for u too.
